### PR TITLE
Adds Event listener for IntelliJ

### PR DESCRIPTION
**Pull request by nspruit**
Refs #2

### CODE REVIEW

File: WatchDogCore/Core.iml [hunk 1]
```diff
@@ -8,7 +8,7 @@
       </configuration>
     </facet>
   </component>
-  <component name="NewModuleRootManager" LANGUAGE_LEVEL="JDK_1_5" inherit-compiler-output="false">
+  <component name="NewModuleRootManager" LANGUAGE_LEVEL="JDK_1_7" inherit-compiler-output="false">
```
Comment (Inventitech) [inline WatchDogCore/Core.iml:11]: Is 1.7 'allowed' by IJ now? @levaja had some concerns about this.

File: WatchDogIntelliJPlugin/WatchDog/src/nl/tudelft/watchdog/intellij/logic/breakpoint/BreakpointCreator.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    public static Breakpoint createBreakpoint(XBreakpoint breakpoint) {
+        Breakpoint result = new Breakpoint(breakpoint.hashCode(),
+                BreakpointClassifier.classify(breakpoint));
+
+        // Initialize enabled and SuspendPolicy fields.
+        result.setEnabled(breakpoint.isEnabled());
+        result.setSuspendPolicy(breakpoint.getSuspendPolicy().ordinal());
+
+        // Initialize hit count field.
+        result.setHitCount(-1);
```
Comment (Inventitech) [inline WatchDogIntelliJPlugin/WatchDog/src/nl/tudelft/watchdog/intellij/logic/breakpoint/BreakpointCreator.java:27]: why with -1 and not 0?

File: WatchDogIntelliJPlugin/WatchDog/src/nl/tudelft/watchdog/intellij/logic/event/listeners/BreakpointListener.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        eventManager.addEvent(event);
+    }
+
+    @Override
+    public void breakpointChanged(@NotNull XBreakpoint xBreakpoint) {
+        Date timestamp = new Date();
+        Breakpoint bp = BreakpointCreator.createBreakpoint(xBreakpoint);
+
+        // Replace entry if present, otherwise create new entry.
+        Breakpoint old = breakpoints.put(bp.getHash(), bp);
```
Comment (Inventitech) [inline WatchDogIntelliJPlugin/WatchDog/src/nl/tudelft/watchdog/intellij/logic/event/listeners/BreakpointListener.java:70]: rename to oldBreakpoint?

File: WatchDogIntelliJPlugin/WatchDog/src/nl/tudelft/watchdog/intellij/logic/event/listeners/BreakpointListener.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        breakpoints.remove(bp.getHash());
+        BreakpointRemoveEvent event = new BreakpointRemoveEvent(bp.getHash(),
+                bp.getBreakpointType(), timestamp);
+        eventManager.addEvent(event);
+    }
+
+    @Override
+    public void breakpointChanged(@NotNull XBreakpoint xBreakpoint) {
+        Date timestamp = new Date();
+        Breakpoint bp = BreakpointCreator.createBreakpoint(xBreakpoint);
```
Comment (Inventitech) [inline WatchDogIntelliJPlugin/WatchDog/src/nl/tudelft/watchdog/intellij/logic/event/listeners/BreakpointListener.java:67]: applies globally: rename bp to breakpoint (whenever I read bp, I have to think of a certain company)?

Comment (nspruit) [inline WatchDogIntelliJPlugin/WatchDog/src/nl/tudelft/watchdog/intellij/logic/event/listeners/BreakpointListener.java:67]: sure

File: WatchDogIntelliJPlugin/WatchDog/src/nl/tudelft/watchdog/intellij/logic/event/listeners/BreakpointListener.java (hunk 3 above)
Comment (nspruit) [inline WatchDogIntelliJPlugin/WatchDog/src/nl/tudelft/watchdog/intellij/logic/event/listeners/BreakpointListener.java:70]: :+1:

File: WatchDogIntelliJPlugin/WatchDog/src/nl/tudelft/watchdog/intellij/logic/breakpoint/BreakpointCreator.java (hunk 2 above)
Comment (nspruit) [inline WatchDogIntelliJPlugin/WatchDog/src/nl/tudelft/watchdog/intellij/logic/breakpoint/BreakpointCreator.java:27]: As -1 indicates that the hit count is disabled in Eclipse, which should be the default value. So, for consistency the hit count is also initialized to -1 to indicate a disabled hit count in IntelliJ.

File: WatchDogCore/Core.iml (hunk 1 above)
Comment (nspruit) [inline WatchDogCore/Core.iml:11]: I think so, as I haven't discovered any issues while compiling locally or on the CI server.

**Pull request merged**